<p>Do not execute an SQL request in a loop</p>
<h2>Noncompliant Code Example</h2>
<pre>

    def foo():
        ...
        baseQuery= "SELECT name FROM users where id = "
        for i in range(0,20):
            query=query + str(i)
            cursor.execute(query) #Noncompliant
            for row in cursor:
                print(row)
        ...
        cursor.close()
-----------------------------------------------------------
    def foo():
        ...
        baseQuery= "SELECT name FROM users where id = "
        data = [ i for i in range(0,20) ]
        cursor.executemany(baseQuery,data)
        for row in cursor:
            print(row)
        ...
        cursor.close()

</pre>


<h2>Compliant Solution</h2>
<pre>

    def  foo() {
        ...
        query = "SELECT name FROM users where id in (0 "
        for i in range(0,20):
            query = query +","+str(i)
        query+=")"
        cursor.execute(query) #compliant

        # iterate through the  resultset
        for row in cursor:
            print(row)

        cursor.close();
        ...
   }

</pre>
