<p>Avoid using relative path</p>
<h2>Noncompliant Code Example</h2>
<pre>
    include 'menu.php'; // NOK
    include './menu.php'; // NOK
    include '../menu.php'; // NOK
    require 'menu.php'; // NOK
</pre>
<h2>Compliant Solution</h2>
<pre>
    include '/menu.php';
</pre>
