<p>Executing SQL queries in loop induced unnecessary network transfert, calculation by the cpu and RAM usage.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    public function foo() {
        ...
        $baseQuery = "SELECT name FROM users where id = ";

        for ($i = 0; $i < 20; ++$i) {

            $query = $baseQuery . $i;
            $connection = mysql_connect($dbhost, $dbuser, $dbpass) or die("Unable to Connect to '$dbhost'");
		        mysql_select_db($dbname) or die("Could not open the db '$dbname'");
            $result = mysql_query($this->Query);// Noncompliant

            // iterate through the result
            ...
            mysql_close($connection);
        }
        ...
    }

</pre>
<h2>Compliant Solution</h2>
<pre>

    public function foo() {
        ...
        $query = "SELECT name FROM users where id in (";

        for ($i = 0; $i < 20; ++$i) {
            $query .= ',' . $i;
        }
        $query .= ')';

        $connection = mysql_connect($dbhost, $dbuser, $dbpass) or die("Unable to Connect to '$dbhost'");
        mysql_select_db($dbname) or die("Could not open the db '$dbname'");
        $result = mysql_query($this->Query); // compliant

        // iterate through the result
        ...
        mysql_close($connection);
   }

</pre>

     <h2>The three sources of impacts of a code identified are: </h2>
    - Energy (measured in joules)
    <br />    - Transfer (measured in Bytes)
    <br />    - Storage (measured in Bytes)<br />
    <br />The control of these 3 impacts allows to lengthen the life of the terminals as well as reduce their energy consumption.
    <br />The ETSdiff tool allows to measure a differential on these three values and in a given context (database and fixed measurement environment).
    <br />The results generated by ETSdiff must help define the interest of the rule reported by Sonarqube in the context of the code analyzed.
    <br />
<img src="https://live.staticflickr.com/65535/52574863536_c912a1cc20_w.jpg"
     alt="ETSdiff percent comparison" style="padding:2rem;">
