Do not call a function when declaring a for-type loop in order to avoid function calls each iteration. It saves CPU cycles.

## Noncompliant Code Example

```python
for i in my_function():  # Noncompliant
    ......
```

## Compliant Solution

```python
limit = my_function()
for i in limit:
    ......
```

