<p>Avoid the use of unoptimized vector images</p>
<h2>Noncompliant Code Example</h2>
<pre>
Votre image contient des métadata non nécessaires à la visualisation de l'image
Your image has metadatas that are not necessary to visualize it

</pre>
<h2>Compliant Solution</h2>
<pre>
Use SVGO to scan your image
</pre>
