:!sectids:

== Why is this an issue?

An app should have a different behavior when device is connected/disconnected to a power station, or has different battery levels.

One can monitor the changes in charging state with a broadcast receiver registered on the actions `ACTION_POWER_CONNECTED` and `ACTION_POWER_DISCONNECTED`, or monitor significant changes in battery level with a broadcast receiver registered on the actions `BATTERY_LOW` and `BATTERY_OKAY`.

== How to fix it

=== Noncompliant code example

```java
// Noncompliant
// No IntentFilter for power charge status
```

=== Compliant solution

```java
new IntentFilter(Intent.ACTION_POWER_CONNECTED)
```

Or

```java
new IntentFilter(Intent.ACTION_POWER_DISCONNECTED)
```

Or

```java
new IntentFilter(Intent.ACTION_POWER_BATTERY_OKAY)
```

Or

```java
new IntentFilter(Intent.ACTION_POWER_BATTERY_LOW)
```

== Resources

- https://github.com/cnumr/best-practices-mobile[Mobile-specific Best Practices] by https://collectif.greenit.fr/index_en.html[CNumR]


