Use `try-with-resources` statement for any object that implements the `AutoCloseable` interface, it save computer resources.

## Noncompliant Code Example

```java
private static void printFileJava7() throws IOException {
    FileInputStream input = new FileInputStream("file.txt");
    int data = input.read();
    while(data != -1){
        System.out.print((char) data);
        data = input.read();
    }
}
```

## Compliant Solution

```java
private static void printFileJava7() throws IOException {
    try(FileInputStream input = new FileInputStream("file.txt")) {
        int data = input.read();
        while(data != -1){
            System.out.print((char) data);
            data = input.read();
        }
    }
}
```
