Avoid using getters and setters in a class, as they increase unnecessary RAM memory usage.

## Noncompliant Code Example

```python
class Client():

    def __init__(self, age):
        self.age = age

    def get_age(self):
        return self.age

    def set_age(self, age):
        self.age = age

client = Client(25)
client.get_age() # Getter inutile
client.set_age(25) # Setter inutile
```

## Compliant Solution

```python
class Client():

    def __init__(self, age):
        self.age = age

    def get_age(self):
        return self.age

    def set_age(self, age):
        self.age = age

client = Client(25)
client.age # RÃ©cupÃ©rer l'attribut age
client.age = 26 # Modifier l'attribut age
```

